### PR TITLE
Improves endpoint handling of descriptors

### LINKED ISSUE #57: [FEATURE] Use BaSyx AAS Web UI with FA³ST Service/Registry
**Describe the solution you'd like**
I would like to use BaSyx AAS Web UI together with an AAS repository and registry from FA³ST (https://github.com/FraunhoferIOSB/FAAAST-Service, https://github.com/FraunhoferIOSB/FAAAST-Registry).
I tried to get it to work, but can't connect from the Web UI to the FA³ST Service/Registry urls, I'm getting network errors.
I guess it's not compatible at the moment, as noted in the documentation: "The AAS Web UI is now only compatible with the components of BaSyx V2 and the Asset Administration Shell V3."

### PR DESCRIPTION
**Pull request by aaronzi**
# Improves endpoint handling of descriptors

## Description of Changes

This PR improves the extraction of the right href from the endpoint of an AAS/Submodel descriptors.
It follows the API specifications in regards to extracting the right endpoint based on the interface-shortName (see screenshot)

image

## Related Issue

Closes #57 

## Additional Information

This fix makes it possible to use the BaSyx AAS Web UI as a frontend for the FA³ST service + registry.
For more details have a look at the issue on the FA³ST repo:
https://github.com/FraunhoferIOSB/FAAAST-Service/issues/914

### CODE REVIEW

File: aas-web-ui/src/components/AppNavigation/AASList.vue [hunk 1]
```diff
@@ -588,7 +591,8 @@
                 let error = false;
                 try {
                     if (this.deleteSubmodels) {
-                        const path = `${this.aasToDelete.endpoints[0].protocolInformation.href}/submodel-refs`;
+                        const shellHref = this.extractEndpointHref(this.aasToDelete, 'AAS-3.0');
+                        const path = `${shellHref}/submodel-refs`;
```
Comment (seicke) [inline aas-web-ui/src/components/AppNavigation/AASList.vue:595]: Just cosmetics (as in the remaining code):

```
const path =  shellHref + '/submodel-refs';
```

**Review by seicke — CHANGES_REQUESTED** [review]

File: aas-web-ui/src/mixins/SubmodelElementHandling.ts [hunk 2]
```diff
@@ -611,5 +613,19 @@ export default defineComponent({
             }
             return sme.idShort ? sme.idShort : '';
         },
+
+        // Extract the right endpoints href from a descriptor
+        extractEndpointHref(descriptor: any, interfaceShortName: string): string {
+            const endpoints = descriptor.endpoints;
```
[the remark this answers is not in the document]
Comment (aaronzi) [reply aas-web-ui/src/mixins/SubmodelElementHandling.ts:619]: Good point 👍🏻

File: aas-web-ui/src/components/AppNavigation/AASListDetails.vue [hunk 3]
```diff
@@ -113,8 +113,9 @@
             // Function to fetch the Asset Details from the AAS Repository
             fetchAssetDetails() {
                 // console.log('fetch asset details: ', this.detailsObject);
-                // strip everything after /shells from detailsobject.endpoints[0].protocolInformation.href
-                let aasRepoEndpoint = this.detailsObject.endpoints[0].protocolInformation.href.split('/shells')[0];
+                const shellHref = this.extractEndpointHref(this.detailsObject, 'AAS-3.0');
+                // strip everything after /shells from the shells href
+                let aasRepoEndpoint = shellHref.split('/shells')[0];
                 let assetInformationEndpoint =
                     aasRepoEndpoint + '/shells/' + this.URLEncode(this.detailsObject.id) + '/asset-information';
```
[the remark this answers is not in the document]
Comment (aaronzi) [reply aas-web-ui/src/components/AppNavigation/AASListDetails.vue:120]: That's a very good find! Thank you 😃

**Review by seicke — APPROVED** [review]
LGTM

**Pull request merged**